File: R/R/l_state_names.R

#' Get State Names of Loon Object
#'
#' States of loon objects can be accessed \code{`[`} and \code{\link{l_cget}}
#' and modified with \code{\link{l_configure}}.
#'
#' @family loon interactive states
#' @template param_target
#'
#' @details
#' In order to access values of a states use \code{\link{l_info_states}}.
#'
#'
#' @return state names
#'
#' @aliases {names.loon}
#'
#' @export
#'
#' @seealso
#' \code{\link{l_info_states}}, \code{\link{l_cget}}, \code{\link{l_configure}}
#'
l_state_names <- function(target) {
    if (!is(target, "loon")) stop("target needs to be of class loon")

    names(l_info_states(target))
}

#' Get State Names of Loon Object
#'
#' States of loon objects can be accessed \code{`[`} and \code{\link{l_cget}}
#' and modified with \code{\link{l_configure}}.
#' @family loon interactive states
#'
#' @param x loon object
#'
#' @return state names
#'
#' @aliases {l_state_names}
#'
#' @export
names.loon <- function(x) {
    l_state_names(x)
}

File: Python/diver/l_state_names.py
from .loon_class import * 
from .l_info import *

def l_state_names(target):
    """Get State Names of Loon Object
    
    Description:
        States of loon objects can be accessed by '[]', '.'and l_cget
        and modified with l_configure.
    
    Args:
        target: target loon class

    Returns:
        state names
    
    Details:
        In order to access values of a states use l_info_states.
    
    SeeAlso: 
        l_info_states, l_cget, l_configure
    
    @namespace loon.l_state_names
    """
    # if(not isinstance(target,loon_class.loon)):
    #     exit('target needs to be of class loon')
    return list(l_info_states(target).keys())
    

